GX-TimeFrequency/src/minimax_omega.F90

```diff
@@ -0,0 +1,3090 @@
+! ***************************************************************************************************
+!  Copyright (C) 2020-2022 Green-X library                                                          
+!  This file is distributed under the terms of the APACHE2 License.                                 
+!                                                                                                   
+! ***************************************************************************************************
+!> \brief This module contains the tabulated minimax coefficients that approximate                  
+!>        1/x ~ 1/pi \sum_{i} w_i x^2/(x^2+a_i^2)^2 with x \in [1:rc]                                     
+!> The arrays containing the coefficients and weights are stored in the `er_aw_aux` derived type.
+!> To extend this module, add the new entries to `omega_npoints_supported`, `energy_ranges_grids`,
+!> and fill the corresponding arrays in the derived type.
+! ***************************************************************************************************
+module minimax_omega
+#include "gx_common.h"
+  use kinds,          only: dp
+  use error_handling, only: register_exc
+  use minimax_utils,  only: bsearch_erange
+  implicit none
+
+  private
+
+  !> list wth the number of points supported for the imag-time meshes
+  integer, parameter :: ngrids = 15
+  integer, parameter, public :: omega_npoints_supported(ngrids) = &
+       [6, 8, 10, 12, 14, 16, 18, 20, 22, 24, 26, 28, 30, 32, 34]
+  integer, parameter :: energy_ranges_grids(ngrids) = &
+       [13, 13, 15, 18, 21, 36, 37, 40, 40, 40, 19, 28, 28, 18, 7]
+
+  type :: er_aw_aux
```

Comment:
check if a separate types module would make sense, i.e, introducing a file minimax_types.F90, with `module  minimax_types` inside, which you then can use where needed

Reply:
That was my intention with the `minimax_utils.F90` module :) I will move it there